### PR TITLE
Generic File Upload

### LINKED ISSUE #2: File upload for any files on the system
It would be great to be able to upload and share any file on the computer.

Considerations:
- maximum file size
- pure image services like Imgur

### LINKED ISSUE #11: Add option to select different uploading services for different filetypes
You might want to upload text to pastie, images to Imgur and everything else to Google Drive so there should be the option to choose which service you use for what.

### PR DESCRIPTION
**Pull request by dtext**
Allows users to upload and share files present on the local file system. The changes in this need to be implemented in #34 and #43, so we should definitely merge this first.
Edit: merged #43 already, order doesn't really matter here.

### Commit Summary
The first commit moves a lot of code from the specific subcommands to the `storage` package. The reasoning behind this is explained in the commit message below.

The second commit just adds the few lines of code that were then needed to implement sharing of existing files.

### Relevant issues
Closes #2, closes #11

### CODE REVIEW

File: src/cli/file.py [hunk 1]
```diff
@@ -1,7 +1,24 @@
...
 
 def main(argv):
-    print(__doc__)
+    args = docopt(__doc__, argv=argv)
+
+    hoster = args["--storage"]
+    file = args["<filename>"]
+
+    if hoster:
+        storage.upload_to(hoster, file)
```
Comment (denniseffing) [inline src/cli/file.py:22]: What happens here if the user enters an invalid combination of hoster and file?

File: src/tools/toolbox.py [hunk 2]
```diff
@@ -32,3 +33,7 @@ def __init__(self):
             raise OSError("Platform not supported!")
         else:
             init_functions[_platform]()
+
+
+def build_filename(media_type: str):
+    return "{}/instantshare_{}_{}.png".format(gettempdir(), media_type, strftime("%Y-%m-%d_%H-%I-%S"))
```
Comment (denniseffing) [inline src/tools/toolbox.py]: Calling `build_filename` with `media_type=text` results in a file named `<tempdir>/instantshare_<media_type>_<timestamp>.PNG` which only makes sense for screenshots.

**Review by denniseffing — CHANGES_REQUESTED** [review]
Solid implementation, even though I dislike the new multi-purpose of our storage module (but it is better than before). I don't have a better solution though, so this has to do for now.

File: src/cli/file.py (hunk 1 above)
Comment (dtext) [reply src/cli/file.py:22]: Any checks of that kind would need to be done inside the different storage modules. We don't really have proper error handling for that yet.

File: src/tools/toolbox.py (hunk 2 above)
Comment (dtext) [reply src/tools/toolbox.py]: That's right. I will see what I can come up with.

**Comment by dtext** [comment]
@denniseffing What do you think about this? We know the file name and extension for every file that is created by us (screenshots, recordings, etc.). For existing files being shared (which could have any file extension), we don't even need to set a filename anyway.

**Review by denniseffing — APPROVED** [review]

**Pull request merged**